RecoHI/HiJetAlgos/plugins/ParticleTowerProducer.cc

```diff
@@ -0,0 +1,208 @@
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Yetkin Yilmaz,32 4-A08,+41227673039,
+//         Created:  Thu Jan 20 19:53:58 CET 2011
+//
+//
+
+#include "DataFormats/CaloTowers/interface/CaloTowerDefs.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerDetId.h"
+#include "DataFormats/Candidate/interface/Particle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/SortedCollection.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/src/WorkerMaker.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "TMath.h"
+
+#include <cmath>
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <vector>
+#include <map>
+#include <utility>
+
+class ParticleTowerProducer : public edm::stream::EDProducer<> {
+public:
+  explicit ParticleTowerProducer(const edm::ParameterSet&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  int eta2ieta(double eta) const;
+  int phi2iphi(double phi, int ieta) const;
+  double ieta2eta(int ieta) const;
+  double iphi2phi(int iphi, int ieta) const;
+  // ----------member data ---------------------------
+
+  CaloGeometry const* geo_;  // geometry
+  edm::EDGetTokenT<reco::PFCandidateCollection> src_;
+  const bool useHF_;
+
+  typedef std::pair<int, int> EtaPhi;
+  typedef std::map<EtaPhi, double> EtaPhiMap;
+  EtaPhiMap towers_;
+
+  // tower edges from fast sim, used starting at index 30 for the HF
+  static constexpr double etaedge[42] = {0.000, 0.087, 0.174, 0.261, 0.348, 0.435, 0.522, 0.609, 0.696, 0.783, 0.870,
+                                         0.957, 1.044, 1.131, 1.218, 1.305, 1.392, 1.479, 1.566, 1.653, 1.740, 1.830,
+                                         1.930, 2.043, 2.172, 2.322, 2.500, 2.650, 2.853, 3.000, 3.139, 3.314, 3.489,
+                                         3.664, 3.839, 4.013, 4.191, 4.363, 4.538, 4.716, 4.889, 5.191};
+  static constexpr int ietaMax = 42;
```

Comment:
use the defined constant:

```suggestion
  static constexpr int ietaMax = 42;
  static constexpr double etaedge[ietaMax] = {0.000, 0.087, 0.174, 0.261, 0.348, 0.435, 0.522, 0.609, 0.696, 0.783, 0.870,
                                         0.957, 1.044, 1.131, 1.218, 1.305, 1.392, 1.479, 1.566, 1.653, 1.740, 1.830,
                                         1.930, 2.043, 2.172, 2.322, 2.500, 2.650, 2.853, 3.000, 3.139, 3.314, 3.489,
                                         3.664, 3.839, 4.013, 4.191, 4.363, 4.538, 4.716, 4.889, 5.191};
```